locking/admin.py

```diff
@@ -116,22 +117,46 @@ def get_urls(self):
         ]
         return locking_urls + urls
 
+    def get_api_url(self, object_id):
+        app_label, model_name = self._model_info
+
+        reverse_kwargs = {
+            'app': app_label,
+            'model': model_name,
+        }
+        if object_id is not None:
+            reverse_kwargs['object_id'] = object_id
+
+        return reverse('locking:locking-api', kwargs=reverse_kwargs)
+
     def get_json_options(self, request, object_id=None):
         app_label, model_name = self._model_info
+
         return json.dumps({
             'currentUser': request.user.username,
             'appLabel': app_label,
+            'apiURL': self.get_api_url(object_id),
             'modelName': model_name,
             'ping': getattr(settings, 'LOCKING_PING_SECONDS', DEFAULT_PING_SECONDS),
-            'objectID': object_id,
+            'messages': {
+                'lockedByMeText': _('You are currently editing this'),
+                'lockedByUserText': _('Locked by'),
+                'takeLockText': _('Take over lock'),
+                'formIsLockedByText': _('Form is locked by'),
+                'lockWasTakenByUserText': _('Another user has taken your lock of this form'),
+                'confirmTakeLockText': _('Are you sure you want to remove this lock?'),
+                'networkWarningText': _('Warning! Due to loss of network connectivity '
+                                        'or a server error, you may not be able '
+                                        'to submit this form.'),
+            },
         })
 
     def locking_admin_form_js(self, request, object_id):
         """Render out JS code for locking a form for a given object_id on this admin"""
         return render(request,
                       'locking/admin_form.js',
                       {'options': self.get_json_options(request, object_id)},
-                      content_type="application/json")
+                      content_type="application/javascript")
```

Comment:
I'd like to try to stick to `application/json` because that's what the JSON spec recommends and that's what Django's built in [JsonResponse object](https://docs.djangoproject.com/en/1.11/ref/request-response/#jsonresponse-objects) uses.

I'm using the `JsonResponse` object elsewhere in this project, so if that isn't causing your problems maybe there is a way to work around this issue? Let me know what problem you are seeing and maybe I can help.